apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/manager/room/RoomManager.kt

```diff
@@ -85,24 +75,31 @@ class RoomManager @Inject constructor(
     private fun cleanUp() {
         room?.disconnect()
         room = null
-        localParticipantWrapper.participant = null
         participants?.clear()
     }
 
+    private fun remoteParticipantCallback(remoteParticipant: RemoteParticipantWrapper) {
+        _onStateEvent.value = RoomViewEvent.OnRemoteParticipantOnClickMenu(remoteParticipant)
+    }
+
     override fun onConnected(room: Room) {
         Timber.i("onConnected -> room sid: %s", room.sid)
 
         room.localParticipant?.let { localParticipant ->
             participants = mutableListOf<ParticipantStream>().let { list ->
-                localParticipantWrapper.participant = localParticipant
+                localParticipantWrapper.localParticipant = localParticipant
+                lifecycleOwner?.lifecycle?.apply { localParticipantWrapper.init(this) }
                 list.add(ParticipantStream(localParticipantWrapper))
 
                 room.remoteParticipants.filter {
                     !it.isVideoComposer()
                 }.forEach {
-                    val newRemoteParticipant = RemoteParticipantWrapper(it)
+                    val newRemoteParticipant = RemoteParticipantWrapper(it, ::remoteParticipantCallback)
+                    lifecycleOwner?.lifecycle?.apply { newRemoteParticipant.init(this) }
+                    newRemoteParticipant.isLocalHost = localParticipantWrapper.isHost
                     list.add(ParticipantStream(newRemoteParticipant))
                 }
+                localParticipantWrapper.publishLocalTracks()
```

Comment:
Curious why now local tracks are published after connected? This will have some effect on UX so I think we want to provide the tracks in the connect options.

Reply:
Done!